File: UpakulERP/UpakulClient/ClientApp/src/app/models/Global/office/office.ts

export interface Office {
    message:string;
    officeId: number;
    officeType: number;
    officeCode: string;
    officeShortCode?: string;
    officeName: string;
    officeAddress: string;
    operationStartDate?: Date; // Store as string for Angular form
    operationEndDate?: Date;
    officeEmail?: string;
    officePhoneNo?: string;
    longitude?: string;
    latitude?: string;
    parentId?: number;

    branchOfficeId: number;


    //officePrincipalId: number;
    //officeZonalId: number;
    // officeRegionalId: number;
    // areaOfficeId: number;

//    regonalOfficeId: number;

    principalOfficeId: number;
    regonalOfficeId: number;
    zonalOfficeId: number;
    areaOfficeId: number;

    // principalOfficeId: number;
    // regonalOfficeId: number;
    // zonalOfficeId: number;

    principalOfficeCode:string;
    principalOfficeName:string;
    zonalOfficeCode:string;
    zonalOfficeName:string;
    regonalOfficeCode:string;
    regonalOfficeName:string;
    areaOfficeCode:string;
    areaOfficeName:string;
    branchOfficeCode:string;
    branchOfficeName:string;
  }

File: UpakulERP/src/Services/Global/Global.Domain/Models/Office.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Utility.Domain;

namespace Global.Domain.Models
{
    [Table("Office", Schema = "dbo")]
    public class Office : EntityBase
    {
        [Key]
        public int OfficeId { get; set; }
        public int OfficeType { get; set; }
        public string OfficeCode { get; set; }
        public string? OfficeShortCode { get; set; }
        public string OfficeName { get; set; }        
        public string? OfficeAddress { get; set; }
        public DateTime? OperationStartDate { get; set; }
        public DateTime? OperationEndDate { get; set; }
        public string? OfficeEmail { get; set; }
        public string? OfficePhoneNo { get; set; }
        public string? Longitude { get; set; }
        public string? Latitude { get; set; }
        public int? ParentId { get; set; }
    }
}
